src/ServiceLayer.Mesh/Functions/FileTransformFunction.cs

```diff
@@ -1,6 +1,50 @@
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using ServiceLayer.Mesh.Data;
+using ServiceLayer.Mesh.Messaging;
+using ServiceLayer.Mesh.Models;
+using ServiceLayer.Mesh.Storage;
+
 namespace ServiceLayer.Mesh.Functions;
 
-public class FileTransformFunction
+public class FileTransformFunction(
+    ILogger<FileTransformFunction> logger,
+    ServiceLayerDbContext serviceLayerDbContext,
+    IMeshFilesBlobStore meshFileBlobStore)
 {
+    [Function("FileTransformFunction")]
+    public async Task Run([QueueTrigger("%FileTransformQueueName%")] FileTransformQueueMessage message)
+    {
+        await using var transaction = await serviceLayerDbContext.Database.BeginTransactionAsync();
+
+        var file = await serviceLayerDbContext.MeshFiles.FirstOrDefaultAsync(f => f.FileId == message.FileId);
+
+        if (file == null)
+        {
+            logger.LogWarning("File with id: {fileId} not found in MeshFiles table.", message.FileId);
+            return;
+        }
+
+        if (file.Status != MeshFileStatus.Extracted)
```

Comment:
Or status is Transforming and it's been stuck for 12 hours... similar logic to the ExtractFunction.
Pull it out info an IsFileSuitableForTransformation method